engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+	inc bc
+	pop hl
+	pop af
+	dec a
+	jr nz, .loop2
+
+	pop af
+	ld [hl], a
+	ret
+
+IntroScene6:
+; Set up grass cutscene (Pikachu/Jigglypuff)
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	call DisableLCD
+	ld b, SGB_GS_INTRO
+	ld c, 1
+	call GetSGBLayout
+	callab InitEffectObject
+	call Intro_ResetLYOverrides
+	ld hl, vChars2
+	ld de, IntroForestGFX
+	call Intro_Copy128Tiles
+	ld a, LOW(Intro_GrassMeta)
+	ld [wIntroTilesPointer + 0], a
+	ld a, HIGH(Intro_GrassMeta)
+	ld [wIntroTilesPointer + 1], a
+	ld hl, vBGMap0
+	ld a, l
+	ld [wIntroBGMapPointer + 0], a
+	ld a, h
+	ld [wIntroBGMapPointer + 1], a
+	ld de, Intro_GrassTilemap + 2 tiles
+	ld a, e
+	ld [wIntroTilemapPointer + 0], a
+	ld a, d
+	ld [wIntroTilemapPointer + 1], a
+	call Intro_DrawBackground
+	ld hl, IntroJigglypuffPikachuGFX
+	ld de, vChars0
+	ld bc, $0a00
+
+.load
+	ld a, [hli]
+	ld [de], a
+	inc de
+	dec bc
+	ld a, b
+	or c
+	jr nz, .load
+
+	ld hl, wSpriteAnimDict
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_OMANYTE
+	ld [hli], a
+	ld a, SPRITE_ANIM_DICT_DEFAULT
+	ld [hli], a
+	xor a
+	ldh [hSCY], a
+	ld [wGlobalAnimYOffset], a
+	ld a, $60
+	ldh [hSCX], a
+	ld a, $a0
+	ld [wGlobalAnimXOffset], a
+	xor a
+	ld [wIntroFrameCounter2], a
+	call EnableLCD
+	ld a, $e4
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	call Intro_InitJigglypuff
+	xor a
+	ld [wIntroSpriteStateFlag], a
+	ret
+
+IntroScene7:
+	call Intro_InitNote
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	inc [hl]
+	and $03
+	ret z
+
+	ld hl, hSCX
+	ld a, [hl]
+	and a
+	jr z, .next
+
+	dec [hl]
+	ld hl, wGlobalAnimXOffset
+	inc [hl]
+	ret
+
+.next
+	ld a, $ff
+	ld [wIntroFrameCounter1], a
+	call Intro_InitPikachu
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ret
+
+IntroScene8:
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	call Intro_InitNote
+	ld hl, wIntroFrameCounter2
+	inc [hl]
+	ret
+
+.next
+	xor a
+	ld [wIntroFrameCounter1], a
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	callab Function9633
+	ret
+
+IntroScene9:
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	swap a
+	and $0f
+	ld e, a
+	ld d, $00
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	and a
+	jr z, .next
+
+	ldh [rBGP], a
+	ld hl, hSCY
+	inc [hl]
+	ld hl, wGlobalAnimYOffset
+	dec [hl]
+	ret
+
+.next:
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ret
+
+.palettes:
+; fade out to black
+	db %11100100
+	db %11100100
+	db %11100100
+	db %11100100
+	db %11100100
+	db %11111001
+	db %11111110
+	db %11111111
+	db 0
+
+Intro_DummyFunction:
+	ret
+
+
+Intro_InitNote:
+	ld a, [wIntroSpriteStateFlag]
+	and a
+	ret nz
+
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	and $3f
+	ret nz
+
+	ld a, [hl]
+	and $7f
+	jr z, .SmallerNote
+
+	depixel 11, 6, 4, 0
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_NOTE
+	call InitSpriteAnimStruct
+	ret
+
+.SmallerNote:
+	depixel 10, 6, 4, 0
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_SMALLER_NOTE
+	call InitSpriteAnimStruct
+	ret
+
+
+Intro_InitJigglypuff:
+	depixel 14, 6
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_JIGGLYPUFF
+	call InitSpriteAnimStruct
+	ret
+
+
```

Comment:
These double blank lines between some functions should just be single lines. Double blank lines are generally just used to separate multiple `SECTION`s in the same file.